src/framework/components/animation/component.js

```diff
@@ -216,12 +217,18 @@ class AnimationComponent extends Component {
     }
 
     /**
-     * Get the duration in seconds of the current animation.
+     * Get the duration in seconds of the current animation. Returns 0 if no animation is playing.
      *
      * @type {number}
      */
     get duration() {
-        return this.animations[this.currAnim].duration;
+        // #if _DEBUG
+        if (!this.currAnim) {
+            Debug.warn(`No animation is playing to get a duration. Returning 0.`);
+        }
+        // #endif
+
+        return this.currAnim ? this.animations[this.currAnim].duration : 0;
```

Comment:
Not much shorter I guess, just trying to avoid ifdefs in general. You could also use Debug.call.
```suggestion
        if (this.currAnim) {
            return this.animations[this.currAnim].duration;
        }

        Debug.warn(`No animation is playing to get a duration. Returning 0.`);
        return 0;
```

Reply:
Oh, I forgot you don't need ifdefs now for messages